rewrite-maven/src/main/java/org/openrewrite/maven/AddAnnotationProcessor.java

```diff
@@ -0,0 +1,104 @@
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.XmlIsoVisitor;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.Optional;
+
+import static org.openrewrite.xml.AddOrUpdateChild.addOrUpdateChild;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddAnnotationProcessor extends Recipe {
+    private static final XPathMatcher PLUGINS_MATCHER = new XPathMatcher("/project/build/plugins");
+    private static final String MAVEN_COMPILER_PLUGIN_GROUP_ID = "org.apache.maven.plugins";
+    private static final String MAVEN_COMPILER_PLUGIN_ARTIFACT_ID = "maven-compiler-plugin";
+
+    @Option(displayName = "Group",
+            description = "The first part of the coordinate 'org.openrewrite.maven:rewrite-maven-plugin:VERSION' of the processor to add.",
+            example = "org.projectlombok")
+    String groupId;
+
+    @Option(displayName = "Artifact",
+            description = "The second part of a coordinate 'org.openrewrite.maven:rewrite-maven-plugin:VERSION' of the processor to add.",
+            example = "lombok-mapstruct-binding")
+    String artifactId;
+
+    @Override
+    public String getDisplayName() {
+        return "Change Maven plugin configuration";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Apply the specified configuration to a Maven plugin. Will not add the plugin if it does not already exist in the pom.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MavenVisitor<ExecutionContext>() {
+            @Override
+            public Xml visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                Xml.Tag plugins = (Xml.Tag) super.visitTag(tag, ctx);
+                if (PLUGINS_MATCHER.matches(getCursor())) {
+                    Optional<Xml.Tag> maybePlugin = plugins.getChildren().stream()
+                            .filter(plugin ->
+                                    "plugin".equals(plugin.getName()) &&
+                                    MAVEN_COMPILER_PLUGIN_GROUP_ID.equals(plugin.getChildValue("groupId").orElse(null)) &&
+                                    MAVEN_COMPILER_PLUGIN_ARTIFACT_ID.equals(plugin.getChildValue("artifactId").orElse(null))
+                            )
+                            .findAny();
+                    if (maybePlugin.isPresent()) {
+                        Xml.Tag plugin = maybePlugin.get();
+                        plugin = new XmlIsoVisitor<ExecutionContext>() {
+                            @Override
+                            public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                                Xml.Tag tg = super.visitTag(tag, ctx);
+                                if (tg.getName().equals("annotationProcessorPaths")) {
```

Comment:
Try to flip these around where possible; we use that pattern elsewhere to avoid null checks.